solr/core/src/java/org/apache/solr/util/PackageTool.java

```diff
@@ -0,0 +1,255 @@
+      log.info("Solr url: "+solrUrl+", solr base url: "+solrBaseUrl);
+      String zkHost = getZkHost(cli);
+
+      log.info("ZK: "+zkHost);
+      String cmd = cli.getArgList().size() == 0? "help": cli.getArgs()[0];
+
+      try (HttpSolrClient solrClient = new HttpSolrClient.Builder(solrBaseUrl).build()) {
+        if (cmd != null) {
+          packageManager = new PackageManager(solrClient, solrBaseUrl, zkHost); 
+          try {
+            repositoryManager = new RepositoryManager(solrClient, packageManager);
+
+            switch (cmd) {
+              case "add-repo":
+                repositoryManager.addRepository(cli.getArgs()[1], cli.getArgs()[2]);
+                break;
+              case "list-installed":
+                packageManager.listInstalled();
+                break;
+              case "list-available":
+                repositoryManager.listAvailable();
+                break;
+              case "list-deployed":
+                if (cli.hasOption('c')) {
+                  String collection = cli.getArgs()[1];
+                  Map<String, SolrPackageInstance> packages = packageManager.getPackagesDeployed(collection);
+                  PackageUtils.printGreen("Packages deployed on " + collection + ":");
+                  for (String packageName: packages.keySet()) {
+                    PackageUtils.printGreen("\t" + packages.get(packageName));                 
+                  }
+                } else {
+                  String packageName = cli.getArgs()[1];
+                  Map<String, String> deployedCollections = packageManager.getDeployedCollections(packageName);
+                  PackageUtils.printGreen("Collections on which package " + packageName + " was deployed:");
+                  for (String collection: deployedCollections.keySet()) {
+                    PackageUtils.printGreen("\t" + collection + "("+packageName+":"+deployedCollections.get(collection)+")");
+                  }
+                }
+                break;
+              case "install":
+              {
+                String packageName = parsePackageVersion(cli.getArgList().get(1).toString())[0];
+                String version = parsePackageVersion(cli.getArgList().get(1).toString())[1];
+                repositoryManager.install(packageName, version);
+                PackageUtils.printGreen(repositoryManager.toString() + " installed.");
+                break;
+              }
+              case "deploy":
+              {
+                String packageName = parsePackageVersion(cli.getArgList().get(1).toString())[0];
+                String version = parsePackageVersion(cli.getArgList().get(1).toString())[1];
+                boolean noprompt = cli.hasOption('y');
+                boolean isUpdate = cli.hasOption("update") || cli.hasOption('u');
+                packageManager.deploy(packageName, version, cli.getOptionValues("collections"), cli.getOptionValues("param"), isUpdate, noprompt);
```

Comment:
Should we sanitize collection names to validate they do not contain `/` which would break the construction of URL later?

Reply:
Done, I've fixed it in a subsequent commit.